apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -48,17 +48,23 @@ type SecretsManagerAPI struct {
 	controllerUUID    string
 	modelUUID         string
 
-	backendConfigGetter commonsecrets.BackendConfigGetter
-	adminConfigGetter   commonsecrets.BackendAdminConfigGetter
-	drainConfigGetter   commonsecrets.BackendDrainConfigGetter
-	remoteClientGetter  func(uri *coresecrets.URI) (CrossModelSecretsClient, error)
+	backendConfigGetter     commonsecrets.BackendConfigGetter
+	adminConfigGetter       commonsecrets.BackendAdminConfigGetter
+	drainConfigGetter       commonsecrets.BackendDrainConfigGetter
+	backendConfigMarshaller func(secretsprovider.ModelBackendConfig) error
+	remoteClientGetter      func(uri *coresecrets.URI) (CrossModelSecretsClient, error)
 
 	crossModelState CrossModelState
 }
 
 // SecretsManagerAPIV1 the secrets manager facade v1.
 // TODO - drop when we no longer support juju 3.1.0
 type SecretsManagerAPIV1 struct {
+	*SecretsManagerAPIV2
+}
```

Comment:
This type should reimplement the methods to do the V1 specific marshalling/compatibility.

Reply:
The marshalling isn't done in the api methods - it's done in the funcs created in register and given to the facade. The facade implementation has no knowledge of the need for this since it just uses the funcs handed to it.

Reply:
I considered changing the logic in the facade such that all the call sites of the various helper methods passed in during register are legacy aware, but that complicates the clean logic in the facade and this way keeps the changes targetted to `register.go` where it's clear what is being done and the compatibility stuff is confined to one place with minimal changes.

Reply:
I've reimplemented the compatibility code on the facade itself.